host_core/lib/host_core/wasmcloud/rpc_invocations.ex

```diff
@@ -0,0 +1,322 @@
+defmodule HostCore.WasmCloud.RpcInvocations do
+  require OpenTelemetry.Tracer, as: Tracer
+  require Logger
+
+  alias HostCore.CloudEvent
+  alias HostCore.Vhost.VirtualHost
+  alias HostCore.WasmCloud.Native
+
+  # Once a message body reaches 900kb, we will use the object
+  # store to hold it and allow 15 seconds for the RPC call to
+  # finish (giving the other side time to "de-chunk")
+  @chunk_threshold 900 * 1024
+  @chunk_rpc_timeout 15_000
+  @rpc_event_prefix "wasmbus.rpcevt"
+
+  # TARGET
+
+  def identify_target(token) do
+    case get_target(token) do
+      :unknown ->
+        %{token | error: "Could not identify suitable target for invocation"}
+
+      target ->
+        %{token | target: target}
+    end
+  end
+
+  defp get_target(%{
+         namespace: namespace,
+         binding: binding,
+         prefix: prefix,
+         source_actor: actor_id
+       }) do
+    case HostCore.Linkdefs.Manager.lookup_link_definition(prefix, actor_id, namespace, binding) do
+      nil ->
+        check_namespace(namespace, prefix)
+
+      ld ->
+        Tracer.set_attribute("target_provider", ld.provider_id)
+        {:provider, ld.provider_id, "wasmbus.rpc.#{prefix}.#{ld.provider_id}.#{binding}"}
+    end
+  end
+
+  defp check_namespace(namespace, prefix) do
+    if String.starts_with?(namespace, "M") && String.length(namespace) == 56 do
+      Tracer.set_attribute("target_actor", namespace)
+      {:actor, namespace, "wasmbus.rpc.#{prefix}.#{namespace}"}
+    else
+      do_lookup_call_alias(namespace, prefix)
+    end
+  end
+
+  defp do_lookup_call_alias(namespace, prefix) do
+    case HostCore.Claims.Manager.lookup_call_alias(prefix, namespace) do
+      {:ok, actor_key} ->
+        Tracer.set_attribute("target_actor", actor_key)
+        {:actor, actor_key, "wasmbus.rpc.#{prefix}.#{actor_key}"}
+
+      :error ->
+        :unknown
+    end
+  end
+
+  # CALL AUTH
+
+  def authorize_call(%{verified: false} = token) do
+    %{token | authorized: false}
+  end
+
+  def authorize_call(
+        %{target: {:provider, _pk, _topic}, claims: claims, namespace: namespace} = token
+      ) do
+    %{token | authorized: Enum.member?(Map.get(claims, :caps, []), namespace)}
+  end
+
+  # allow actor-to-actor calls
+  def authorize_call(%{target: {:actor, _, _}} = token) do
+    %{token | authorized: true}
+  end
+
+  # LINKS
+
+  # Built-in Providers do not have link definitions
+  # their implementations are already handled either in the NIF or in wasmCloud runtime
+
+  # default behavior is to allow actor-to-actor calls
+  def verify_link(%{target: {:actor, _, _}} = token) do
+    %{token | verified: true}
+  end
+
+  def verify_link(
```

Comment:
`verify_link` needs a branch that can handle the target not being found in `identify_target`, or else we get a nasty crash like 
```
            ** (FunctionClauseError) no function clause matching in HostCore.WasmCloud.RpcInvocations.verify_link/1
                (host_core 0.62.0) lib/host_core/wasmcloud/rpc_invocations.ex:87: HostCore.WasmCloud.RpcInvocations.verify_link(%{<blahblahtokenstuff>, error: "Could not identify suitable target for invocation", 
```

Reply:
This is because the `target` field isn't present